app/models/issue_tta_report.rb

```diff
@@ -1,5 +1,54 @@
+require "api_delegator"
+
 class IssueTtaReport < ApplicationRecord
-  validates :issue_id, :tta_report_display_id, :tta_report_id, presence: true
-  validates :tta_report_id, format: {with: /\A[0-9]*\z/, message: "only allows numbers"}
+  validates :issue_id, :tta_report_display_id, presence: true
   validates :tta_report_display_id, uniqueness: {scope: :issue_id}
+  validate :api_call_succeeded
+  validates :tta_report_id, presence: true, if: :errors_blank?
+
+  attr_accessor :access_token
+
+  before_validation :retrieve_tta_report_id
+
+  private
+
+  def api_call_succeeded
+    # do not make the api call here, if @activity_data is nil then we didn't attempt to make a call
+    if !@activity_data.nil?
```

Comment:
[thought] mmmm the conditionals moved but they're still here. but creating a response class it def outside the scope, just a thing to think about for future PRs imo.

however
[suggestion] this could be condensed to
`if activity_data&[:success]` to at least cut down on the double ifs